datalad/customremotes/main.py

```diff
@@ -77,7 +78,10 @@ def _main(args, backend=None):
     assert(backend is not None)
     if backend == 'archive':
         from .archives import ArchiveAnnexCustomRemote
-        remote = ArchiveAnnexCustomRemote()
+        autoclean = cfg.obtain("datalad.remote.archives.autoclean")
+        remote = ArchiveAnnexCustomRemote(
+            persistent_cache=not autoclean
```

Comment:
although I do like `autoclean`, it is not really corresponding to the underlying mechanism, smth like `cache.persistent = bool` would have been more reflecting of what is going on.

gave me also idea for `autodrop` where we would also auto drop the underlying archive if that process did `get` it (not that it should be done here, just an idea).  But with the comment below on `jobs` -- for `autodrop` might be even trickier.

Reply:
Yes, wondered about better naming as well, just went with what you proposed in the issue for now.

> But with the comment below on jobs -- for autodrop might be even trickier.

Yes, I'm not sure how to address this. The special remote itself might be the wrong place.

Reply:
> Yes, I'm not sure how to address this. The special remote itself might be the wrong place.

may be you are hitting it on the nail -- may be it could be `datalad.runtime.get.autoclean` which just calls out to `clean` upon completion, and thus taking care about cleaning everything (archives cache and also possibly some annex tmp files etc)?

Reply:
oh no -- also a tricky case: I remember having nested tarballs !  Then we might end up in some nested `datalad.get` which would then react on `autoclean` and clean up after itself **and** parent process... not good.  tricky business it is all this "cleaning" ;)

Reply:
> may be it could be datalad.runtime.get.autoclean which just calls out to clean upon completion, and thus taking care about cleaning everything (archives cache and also possibly some annex tmp files etc)?

Currently thinking in the same direction, yes.

>  Then we might end up in some nested datalad.get which would then react on autoclean and clean up after itself and parent process... not good.

Hm. What about changing the cache dir layout? `.git/datalad/tmp/archives/<key of the archive itself>/...`? Then a get theoretically could clean only the for the key(s) it was supposed to get. May need some more serious RF'ing then it sounds, though.